daemon/cmd/devices.go

```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2016-2021 Authors of Cilium
+
+// This module implements Cilium's network device detection.
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/probes"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/mac"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	excludedDevicePrefixes = []string{
+		"cilium_",
+		"lo",
+		"lxc",
+		"cni",
+		"docker",
+	}
+)
+
+type DeviceManager struct {
+	mu      lock.Mutex
+	devices map[string]bool
```

Comment:
It's the suggested way of making a set in go: https://golang.org/doc/effective_go#maps. The boolean just marks existence.
I've seen "struct{}" being used elsewhere. Is there a reason to prefer that (I find it much uglier)?

Reply:
I personally like `map[string]void` but I don't think we have a precedent in the codebase. Regardless of what we use, could you add a comment explaining it's a set. It's not immediately obvious otherwise.

Reply:
Will add, thanks!

Reply:
I'll switch this to `struct{}` as that's the normal pattern.

Reply:
> I've seen "struct{}" being used elsewhere

AFAIK `struct{}` should not allocate any mem for val, while `bool` does.